pennylane/queuing.py

```diff
@@ -87,6 +88,56 @@ def active_context(cls):
         """Returns the currently active queuing context."""
         return cls._active_contexts[-1] if cls.recording() else None
 
+    @classmethod
+    @contextmanager
+    def stop_recording(cls):
+        """A context manager and decorator to ensure that contained logic is non-recordable
+        or non-queueable within a QNode or quantum tape context.
+
+        **Example:**
```

Comment:
These examples feel awkward to me because it doesn't really seem like a scenario in which I'd actually want to stop recording, whereas the old examples stopped recording in a function that generated a list (as an.. example)

Reply:
Sure. I can improve the documentation.

Reply:
I was trying to make it more concise, but I can see how it then also gets more confusing.